stream/kafka/kafka_test.go

```diff
@@ -0,0 +1,215 @@
+package kafka
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"math/rand"
+	"strconv"
+	"testing"
+	"time"
+
+	"github.com/confluentinc/confluent-kafka-go/kafka"
+	"github.com/google/go-cmp/cmp"
+
+	"github.com/adevinta/graph-vulcan-assets/stream"
+	"github.com/adevinta/graph-vulcan-assets/stream/streamtest"
+)
+
+const (
+	bootstrapServers = "127.0.0.1:9092"
+	groupPrefix      = "stream_kafka_kafka_test_group_"
+	topicPrefix      = "stream_kafka_kafka_test_topic_"
+	messagesFile     = "testdata/messages.json"
+	timeout          = 5 * time.Minute
+)
+
+func init() {
+	rand.Seed(time.Now().UnixNano())
+}
+
+func setupKafka(topic string) (msgs []stream.Message, err error) {
+	cfg := &kafka.ConfigMap{
+		"bootstrap.servers": bootstrapServers,
+
+		// Set message timeout to 5s, so the kafka client returns an
+		// error if the broker is not up.
+		"message.timeout.ms": 5000,
+	}
+
+	prod, err := kafka.NewProducer(cfg)
+	if err != nil {
+		return nil, fmt.Errorf("error creating producer: %w", err)
+	}
+	defer prod.Close()
+
+	msgs = streamtest.Parse(messagesFile)
+	for _, msg := range msgs {
+		if err := produceMessage(prod, topic, msg); err != nil {
+			return nil, fmt.Errorf("error producing message: %w", err)
+		}
+	}
+
+	for prod.Flush(10000) > 0 {
```

Comment:
Same question as the one I made for the ``main_test.go`` file in same the function: is calling to the ``Flush`` method really needed?

Reply:
I think it is not. Done.